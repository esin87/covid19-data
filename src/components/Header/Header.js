import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { LinkContainer } from 'react-router-bootstrap';

const Header = () => {
	return (
		<Navbar variant='light' expand='md' style={{ marginBottom: '2.5rem' }}>
			<LinkContainer to='/home'>
				<Navbar.Brand>COVID-19 Data</Navbar.Brand>
			</LinkContainer>
			<Navbar.Toggle aria-controls='basic-navbar-nav' />
			<Navbar.Collapse id='basic-navbar-nav'>
				<Nav>
					<LinkContainer to='/home'>
						<Nav.Link>Home</Nav.Link>
					</LinkContainer>
					<LinkContainer to='/countries'>
						<Nav.Link>Country List</Nav.Link>
					</LinkContainer>
					<LinkContainer to='/search'>
						<Nav.Link>Country Search</Nav.Link>
					</LinkContainer>
					<LinkContainer to='/global'>
						<Nav.Link>Global Summary</Nav.Link>
					</LinkContainer>
					<LinkContainer to='/about'>
						<Nav.Link>About</Nav.Link>
					</LinkContainer>
				</Nav>
			</Navbar.Collapse>
		</Navbar>
	);
};

export default Header;
