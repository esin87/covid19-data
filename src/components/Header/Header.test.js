import React from 'react';
import Header from './Header';
import { HashRouter } from 'react-router-dom';
import { mount, render } from 'enzyme';

describe('Header component', () => {
	let component;

	beforeEach(() => {
		component = mount(
			<HashRouter>
				<Header />
			</HashRouter>
		);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('should have a first anchor tag with appropriate text', () => {
		expect(component.find('a').at(0).text()).toBe('COVID-19 Data');
	});

	it('should have a second anchor tag with appropriate text', () => {
		expect(component.find('a').at(1).text()).toBe('Home');
	});

	it('should have a third anchor tag wtih appropriate text', () => {
		expect(component.find('a').at(2).text()).toBe('Country List');
	});
});
