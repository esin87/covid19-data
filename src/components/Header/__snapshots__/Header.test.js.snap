// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match the snapshot 1`] = `"<nav style=\\"margin-bottom: 2.5rem;\\" class=\\"navbar navbar-expand-md navbar-light\\"><a href=\\"#/home\\" class=\\"navbar-brand\\">COVID-19 Data</a><button aria-controls=\\"basic-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"navbar-collapse collapse\\" id=\\"basic-navbar-nav\\"><div class=\\"navbar-nav\\"><a href=\\"#/home\\" data-rb-event-key=\\"#/home\\" class=\\"nav-link\\">Home</a><a href=\\"#/countries\\" data-rb-event-key=\\"#/countries\\" class=\\"nav-link\\">Country List</a><a href=\\"#/search\\" data-rb-event-key=\\"#/search\\" class=\\"nav-link\\">Country Search</a><a href=\\"#/global\\" data-rb-event-key=\\"#/global\\" class=\\"nav-link\\">Global Summary</a><a href=\\"#/about\\" data-rb-event-key=\\"#/about\\" class=\\"nav-link\\">About</a></div></div></nav>"`;
